.github/scripts/libmongocrypt.mjs

```diff
@@ -0,0 +1,125 @@
+import util from 'node:util';
+import process from 'node:process';
+import fs from 'node:fs/promises';
+import child_process from 'node:child_process';
+import events from 'node:events';
+import path from 'node:path';
+
+async function parseArguments() {
+  const jsonImport = { [process.version.split('.').at(0) === 'v16' ? 'assert' : 'with']: { type: 'json' } };
+  const pkg = (await import('../../package.json', jsonImport)).default;
+  const libmongocryptVersion = pkg['mongodb:libmongocrypt'];
+
+  const options = {
+    url: { short: 'u', type: 'string', default: 'https://github.com/mongodb/libmongocrypt.git' },
+    libversion: { short: 'l', type: 'string', default: libmongocryptVersion },
+    clean: { short: 'c', type: 'boolean' },
+    help: { short: 'h', type: 'boolean' }
+  };
+
+  const args = util.parseArgs({ args: process.argv.slice(2), options, allowPositionals: false });
+
+  if (args.values.help) {
+    console.log(
+      `${process.argv[1]} ${[...Object.keys(options)]
+        .filter(k => k !== 'help')
+        .map(k => `[--${k}=${options[k].type}]`)
+        .join(' ')}`
+    );
+    process.exit(0);
+  }
+
+  return {
+    libmongocrypt: { url: args.values.url, ref: args.values.libversion },
+    clean: args.values.clean
+  };
+}
+
+/** `xtrace` style command runner, uses spawn so that stdio is inherited */
+async function run(command, args = [], options = {}) {
+  console.error(`+ ${command} ${args.join(' ')}`, options.cwd ? `(in: ${options.cwd})` : '');
+  await events.once(child_process.spawn(command, args, { stdio: 'inherit', ...options }), 'exit');
+}
+
+/** CLI flag maker: `toFlags({a: 1, b: 2})` yields `['-a=1', '-b=2']` */
+function toFlags(object) {
+  return Array.from(Object.entries(object)).map(([k, v]) => `-${k}=${v}`);
+}
+
+const args = await parseArguments();
+const libmongocryptRoot = path.resolve('_libmongocrypt');
+
+const currentLibMongoCryptBranch = await fs.readFile(path.join(libmongocryptRoot, '.git', 'HEAD'), 'utf8').catch(() => '')
+const libmongocryptAlreadyClonedAndCheckedOut = currentLibMongoCryptBranch.trim().endsWith(`r-${args.libmongocrypt.ref}`);
+
+if (!args.clean && !libmongocryptAlreadyClonedAndCheckedOut) {
+  console.error('fetching libmongocrypt...', args.libmongocrypt);
+  await fs.rm(libmongocryptRoot, { recursive: true, force: true });
+  await run('git', ['clone', args.libmongocrypt.url, libmongocryptRoot]);
+  await run('git', ['fetch', '--tags'], { cwd: libmongocryptRoot });
+  await run('git', ['checkout', args.libmongocrypt.ref, '-b', `r-${args.libmongocrypt.ref}`], { cwd: libmongocryptRoot });
+} else {
+  console.error('libmongocrypt already up to date...', args.libmongocrypt);
+}
+
+const libmongocryptBuiltVersion = await fs.readFile(path.join(libmongocryptRoot, 'VERSION_CURRENT'), 'utf8').catch(() => '');
+const libmongocryptAlreadyBuilt = libmongocryptBuiltVersion.trim() === args.libmongocrypt.ref;
+
+if (!args.clean && !libmongocryptAlreadyBuilt) {
```

Comment:
```suggestion
if (args.clean || !libmongocryptAlreadyBuilt) {
```

Am I misunderstanding clean?  I assumed that when clean is true, it force installs and builds libmongocrypt

Reply:
(same comment for the other if-statement as well)

Reply:
Fixed!